### DISCUSSION TITLE
Persistent sendBlockChanges method

### DISCUSSION THREAD

**Discussion by bkgktp** [PurpurMC/Purpur#1588] [Feature Requests]
I want to overlay a note blog on a server-sided stone block and I need to make it permanent. I would like to add a persistence setting to this method so that we can change the behavior of the blocks client-side and server-side. this would also work very, very well for servers using resource-pack because it would be nice to make sure that data sent client-side doesn't change no matter what. I tried to do this as a plugin, but this is insufficient, it definitely needs to be added to the server fork and everyone should benefit from it. there should be a setting to not accept any block updates in the block with the persistent sendBlockChanges method applied, or a setting for this should be added. this is quite doable, even if the setting is only for notablocks and leaves, this is acceptable.

for a player with a client-side block change, this breaks when the player changes parts, leaves and enters the server, or there is a physical update to the block, and I can't synchronize it no matter what. this should be added to purpurmc in a more perofessional way, so we can further improve the note blocks or use them in custom applications. it also allows us to make custom leaves with this resource pack.

as an example you can see this in the new version of the HMCLeaves plugin, this is trying to achieve the impossible, but coding this feature with a plugin causes a lot of problems, adding it to the server fork will make it more performant.

**Comment by Onako2** [comment]
Saving stuff that are meant for the client across server restarts isn't a good idea because you would need a replica of everything that has been sent by plugins. What if some other plugin wants to edit the block like plugins are doing already? That should be implemented at most in Paper, not in Purpur because Purpur stuff isn't as widely adopted as Paper stuff
